streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -109,11 +109,6 @@ public StreamTask(int id,
 
         log.info("Creating restoration consumer client for stream task [" + id + "]");
 
-        Consumer restoreConsumer = new KafkaConsumer<>(
-                config.getConsumerConfigs(),
-                new ByteArrayDeserializer(),
-                new ByteArrayDeserializer());
-
```

Comment:
For testability purposes, its important to be able to pass in `MockConsumer` and  `MockProducer` instances rather than hard-code the creation of them.